spec/message.abnf

```diff
@@ -0,0 +1,57 @@
+message = [s] *(declaration [s]) body [s]
+
+declaration = let s variable [s] "=" [s] "{" [s] expression [s] "}"
+body = pattern
+     / (selectors 1*([s] variant))
+
+pattern = "{" *(text / placeholder) "}"
+selectors = match 1*([s] selector)
+selector = "{" [s] expression [s] "}"
+variant = when 1*(s key) [s] pattern
+key = nmtoken / literal / "*"
+
+placeholder = "{" [s] (expression / markup / markup-end) [s] "}"
+
+expression = ((literal / variable) [s annotation])
+           / annotation
+annotation = function *(s option)
+option = name [s] "=" [s] (literal / nmtoken / variable)
+
+markup = markup-start *(s option)
+
+let = %x6C.65.74
+match = %x6D.61.74.63.68
+when = %x77.68.65.6E
```

Comment:
It felt like a good idea to not require RFC 7405 just for 3 keywords.

Reply:
Can we avoid the hex, and just have:

let = "let"
match = "match"
when = "when"

Far more readable, and less error prone (how many people would see that let = %x6C.65.75 is wrong??)

Reply:
My understanding is that `let = "let"` would make the `let` keyword case-insensitive, due to how literal text strings work in ABNF. That's why we'd need the case-sensitive extension: `let = %s"let"`. See also: https://github.com/unicode-org/message-format-wg/pull/347/files#diff-a33e1c859f2aaad86b37ea3b3b5e8d45331199d6e879c5aba38eca2f23f01865.

Reply:
This can merit a comment in the ABNF to prevent folks wondering, e.g.

```abnf
; reserved keywords are always lowercase
let   = %x6C.65.74        ; "let"
match = %x6D.61.74.63.68  ; "match"
when  = %x77.68.65.6E     ; "when"
```

Reply:
Add the comment suggested by @aphillips in 4205cd2.